src/vs/workbench/contrib/externalTerminal/browser/externalTerminal.contribution.ts

```diff
@@ -26,114 +26,195 @@ import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle
+				const config = configurationService.getValue<IExternalTerminalConfiguration>();
+				const useIntegratedTerminal = remoteAgentService.getConnection() || explorerKind === 'integrated';
+				const targets = distinct(stats.filter(data => data.success));
+				if (useIntegratedTerminal) {
+					// TODO: Use uri for cwd in createterminal
+					const opened: { [path: string]: boolean } = {};
+					const cwds = targets.map(({ stat }) => {
+						const resource = stat!.resource;
+						if (stat!.isDirectory) {
+							return resource;
+						}
+						return URI.from({
+							scheme: resource.scheme,
+							authority: resource.authority,
+							fragment: resource.fragment,
+							query: resource.query,
+							path: dirname(resource.path)
+						});
 					});
-				});
-				for (const cwd of cwds) {
-					if (opened[cwd.path]) {
-						return;
-					}
-					opened[cwd.path] = true;
-					const instance = await integratedTerminalService.createTerminal({ config: { cwd } });
-					if (instance && instance.target !== TerminalLocation.Editor && (resources.length === 1 || !resource || cwd.path === resource.path || cwd.path === dirname(resource.path))) {
-						integratedTerminalService.setActiveInstance(instance);
-						terminalGroupService.showPanel(true);
+					for (const cwd of cwds) {
+						if (opened[cwd.path]) {
+							return;
+						}
+						opened[cwd.path] = true;
+						const instance = await integratedTerminalService.createTerminal({ config: { cwd } });
+						if (instance && instance.target !== TerminalLocation.Editor && (resources.length === 1 || !resource || cwd.path === resource.path || cwd.path === dirname(resource.path))) {
+							integratedTerminalService.setActiveInstance(instance);
+							terminalGroupService.showPanel(true);
+						}
 					}
+				} else if (externalTerminalService) {
+					distinct(targets.map(({ stat }) => stat!.isDirectory ? stat!.resource.fsPath : dirname(stat!.resource.fsPath))).forEach(cwd => {
+						externalTerminalService!.openTerminal(config.terminal.external, cwd);
+					});
 				}
-			} else if (externalTerminalService) {
-				distinct(targets.map(({ stat }) => stat!.isDirectory ? stat!.resource.fsPath : dirname(stat!.resource.fsPath))).forEach(cwd => {
-					externalTerminalService!.openTerminal(config.terminal.external, cwd);
-				});
-			}
-		});
-	}
-});
+			});
+		}
+	});
+}
+
+registerOpenTerminalCommand(OPEN_IN_INTEGRATED_TERMINAL_COMMAND_ID, 'integrated');
+registerOpenTerminalCommand(OPEN_IN_EXTERNAL_TERMINAL_COMMAND_ID, 'external');
 
 export class ExternalTerminalContribution extends Disposable implements IWorkbenchContribution {
-	private _openInTerminalMenuItem: IMenuItem;
+	private _openInIntegratedTerminalMenuItem: IMenuItem;
+	private _openInExternalTerminalMenuItem: IMenuItem;
+
+	private _IntegratedOpenEditors?: IDisposable;
+	private _IntegratedExplorer?: IDisposable;
+
+	private _ExternalOpenEditors?: IDisposable;
+	private _ExternalExplorer?: IDisposable;
```

Comment:
Lowercase `_i...` and `_e...` is preferred

Reply:
I was negligent